008-auth.md

```diff
@@ -0,0 +1,254 @@
+outcome?
+
+# Assumptions
+
+What assumptions do we make about the elements this RFC touches?
+
+If this is a feature with significant memory overhead, do we assume end-users
+can all meet it?
+
+If this RFC is to make a security improvement, what threat model is it
+operating under? List any assumptions we make about the capabilities we feel
+we can reasonably make about potential adversaries.
+
+# Requirements
+
+What exactly is the outcome we want and want to enforce moving forward?
+
+List some MUST, MUST NOT, and SHOULD statements here that people will be
+expected to use as a quick reference to consider when doing engineering work.
+
+# Design
+
+This should be a simple bulleted explanation of the core process or outcome
+that we expect when this RFC is implemented.
+
+This could be a developer workflow, an automation workflow, a specific sequence
+of events, etc.
+
+Include any diagrams as appropriate, favoring text-based sources others can
+easily edit like mermaid, UML, etc. where possible, with external files sharing
+the same filename prefix as this document.
+
+# Examples
+
+Detail one or more hypothetical but plausible real-world scenarios where this
+RFC would offer a tangible benefit.
+
+# Drawbacks
+
+Provide a few bullets honestly detailing the downsides of implementing this.
+
+Will it slow down development? Cost more resources? Etc.
+
+# Alternatives
+
+List any alternatives others have tried, including the current solution if
+there is one, to help others quickly catch up on how we got here.
+
+# Scope
+
+List a few items that are clearly in scope, anything that we want to do that is
+explicitly deferred to future work.
+
+## Questions
+
+What open questions still exist that we can't answer right now?
+
+
+
+
+
+
+# Client authentication
+
+
+## Phase 1
+
+First, we need to break apart the `talosconfig` from the authentication PKI.
+This allows us to, among other things: 
+ - encourage the distribution of `talosconfig` without secret material included
+ - encourage the disuse of root admin keys for normal operation
+ - encourage the use of user-specific secret material which is never sent over
+   the wire
+ - use a common user keypair across multiple Talos clusters
+
+Some minor additions to `talosconfig` will be useful:
+ - top-level `defaultAuthIdentity` reference
+ - cluster-level `authIdentity` reference
+ - cluster-level `clusterId` reference
+ - cluster-level `certs` list
+
+Perhaps add a `-i` option to `talosctl` which tells Talos to use the
+given auth identity.
+
+Construct and use `~/.config/talos/auth` (or ~/.talos/auth, for non-XDG-standard
+machines) to house the Talos identities on the client machine.
+
+Modify `talosctl` to use the search the auth directory to resolve authentication
+material when connecting to Talos.
+
+Proposed structure of `~/.config/talos/auth`:
+ - `id_<name>.key` : private key of the given id
+ - `id_<name>.pub` : public key of the given id
+
+Proposed CLI subcommand:  `auth`:
```

Comment:
This smells a lot like " we want to keep holding the CA inside the cluster, and introduce a complicated management interface around it".

I'm not sure I like that, and I'm not confident it'd be very automate-able.

What about going the other way round, having the CA outside of Talos, and passing individual certs to it?
Rotations would need to be manual, or there some automated rotation mechanism (think of ACME).